add signalr client feature matrix (#14394)
aspnet/AspNetCore.Docs: aspnetcore/signalr/client-features.md

---

> title: SignalR Client Features

Reviewer: ```suggestion
title: SignalR client features
```

author: bradygaster

> description: Features as supported by the various ASP.NET Core SignalR clients.

Reviewer: ```suggestion
description: Learn which features are supported by the various ASP.NET Core SignalR clients.
```

monikerRange: '>= aspnetcore-3.0'
ms.author: bradyg
ms.custom: mvc
ms.date: 09/18/2019
uid: signalr/client-features
---
# ASP.NET Core SignalR client features

## Feature distribution

> The table below illustrates the matrix of features and their support across the clients that offer real-time support to various platforms. 

Reviewer 2: The table below shows the features and support for the clients that offer real-time support.

Reviewer 2: Less is more. My version will MT much better.

| Feature | .NET Core | JavaScript | Java |
| ---- | :-: | :-: | :-: |

> | [Server-to-client Streaming](xref:signalr/streaming)          |✔|✔| |

Reviewer 3: ```suggestion
| [Server-to-client Streaming](xref:signalr/streaming)          |✔|✔|✔|
```

Reviewer 4: Do we even need a table then?

Reviewer 4: Nvm. I guess we keep it for auto-reconnect.

Reviewer 3: Also we'll probably be adding more rows

Author: also think about things like long polling, sse - how are those supported across all the clients. **absolutely** just isn't about "new features" as much as it is answering the perpetual "what can i do where" question.

> | [Client-to-server Streaming](xref:signalr/streaming)          |✔|✔| |

Reviewer 3: ```suggestion
| [Client-to-server Streaming](xref:signalr/streaming)          |✔|✔|✔|
```

> | Automatic Reconnection ([.NET](/aspnet/core/signalr/dotnet-client?view=aspnetcore-3.0&tabs=visual-studio#handle-lost-connection), [JavaScript](/aspnet/core/signalr/javascript-client?view=aspnetcore-3.0#reconnect-clients))          |✔|✔| |

Reviewer 5: Does it make sense to include a reference to the issue tracking this work? I'd love for customers that come to this page to sound off and let us know how important it is that we close this feature gap. It would help in prioritization :)

https://github.com/aspnet/AspNetCore/issues/8711 is the issue.

Reviewer 5: I'll move this discussion to my new PR to update this doc further.

Reviewer 3: There is also this issue https://github.com/aspnet/AspNetCore.Docs/issues/14378 where we started listing things

Reviewer 5: Will update my PR from that list.

## Additional resources

* [Get started with SignalR for ASP.NET Core](xref:tutorials/signalr)

> * [Supported Platforms](xref:signalr/supported-platforms)

Reviewer: ```suggestion
* [Supported platforms](xref:signalr/supported-platforms)
```
